src/software/parser.cpp

```diff
@@ -401,7 +406,10 @@ String parseCommand(String buffer) {
 
                 // Sanitize the inputs
                 if (rate <= 0) {
-                    rate = DEFAULT_STEPPING_RATE;
+                    rate = motor.gm_code.rate;
+                }
+                else {
```

Comment:
It shouldn't reset the default rate every time. scheduleSteps will handle the rates

Reply:
https://marlinfw.org/docs/gcode/G006.html
<html>
<body>
<!--StartFragment-->

[R<rate>] | Step rate per second. Last value is cached for future invocations.
-- | --


<!--EndFragment-->
</body>
</html>

It will use the previous rate value

```
                int32_t rate = parseValue(buffer, 'R').toInt();
...
                if (rate <= 0) {
                    rate = motor.gm_code.rate;
                }
                else {
                    motor.gm_code.rate = rate;
                }
```